Pidaq/Lib/can_manager/can_manager.py

```diff
@@ -0,0 +1,171 @@
+"""Contains classes and methods to implement and manage the can (socketcan) bus
+
+Classes:
+    CanManager
+    SensorReading
+
+Functions:
+    parse_conversion_factor
+"""
+import json
+import os
+
+import can
+
+
+class CanManager:
+    """Manages the can bus, and sets the SensorReading objects for the test being run
+
+    This class contains the socketcan bus instance that will be used to communicate with
+    the sensor board, as well as the configured sensor readings for the particular project
+    that is being tested
+
+    Attributes:
+        bus (can.interfaces.socketcan.SocketcanBus)
+        messages (list(SensorReading)): list of all the messages to be expected from the 
+                                        sensor board
+        message_ids (list(int)): list of all the message ids, hexidecimal integers
+
+    Methods:
+        read_message_config(project: str)
+            Reads the configuration of sensor readings from messageconfig.json
+        send_message(id: int, data: list)
+        read_bus
+            Reads the bus for a message
+        assign_message_data(bus_message: can.Message)
+            assigns data to the correct SensorReading object
+    """
+    def __init__(self, bus_name: str) -> None:
+        self.bus = can.interfaces.socketcan.SocketcanBus(channel=bus_name)
+        self.messages = []
```

Comment:
As I put this up I was thinking it might be better to store the SensorReading objects in a dict, with the key being the reading name, thoughts?

Reply:
Yeah I like it, key search instead of searching an array